### PR TITLE
Update of #1165 (add site clock to simulation) to fix BIPM

### LINKED ISSUE #1464: Simulated TOAs are not correctly written into a tim file
The `pint.simulation.make_fake_toas_uniform` function generates uniformly spaced TOAs. When I plot the residuals of the simulated `TOAs` object, I get zero residuals up to some numerical error as expected. However, some modulations are seen when I write this `TOAs` object into a `tim` file and then plot the residuals from the generated tim file. 

Why does this happen? 

Here is some code to reproduce this. The difference is at the ~100 ns level.
```
from pint.config import examplefile
from pint.simulation import make_fake_toas_uniform
from pint.models import get_model, get_model_and_toas
from pint.residuals import Residuals
import astropy.units as u
import matplotlib.pyplot as plt

parfile = examplefile("NGC6440E.par")
timfile = "fake.tim"

model = get_model(parfile)

toas = make_fake_toas_uniform(
        startMJD=56000,
        endMJD=56400,
        ntoas=100,
        model=model,
        obs="ao",
        error=1 * u.microsecond,
        freq=1400 * u.MHz
    )
res = Residuals(toas, model)

plt.plot(toas.get_mjds(), res.calc_time_resids(), label="From TOAs object (before writing)")

toas.write_TOA_file(timfile, name="fake", format="TEMPO2")

model1, toas1 = get_model_and_toas(parfile, timfile)
res1 = Residuals(toas1, model1)

plt.plot(toas1.get_mjds(), res1.calc_time_resids().to("us"), label="From tim file (after writing)")
plt.xlabel("MJD")
plt.ylabel("Residuals (us)")
plt.legend()
plt.show()
```

EDIT: Here is the residual plot.

### PR DESCRIPTION
**Pull request by dlakaplan**
This updates #1165 to fix #1464 and #1161. That separately implemented a check as to whether the clock was OK, but didn't set `include_bipm` correctly when needed. This updates that and adds a check of residuals roundtrip through a file at higher precision.

### CODE REVIEW

**Comment by dlakaplan** [comment]
So far this passes the RTD CI.  It doesn't have all of the updated tests that #1165 had: I will try to add those back.

**Comment by dlakaplan** [comment]
Actually, I'm not sure those other tests are needed.   They mostly added roundtrip tests on top of others, but we have a separate roundtrip test here.  So I think this should be sufficient.

**Comment by dlakaplan** [comment]
Separately I was wondering if some of those methods should also be able to operate on `astropy.Time` objects, and not just TOAs.  This might make for a simpler interface for some use cases, and the difference between a TOA and a more generic Time is almost never relevant for DM calculations.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ and this project, at least loosely, adheres to [Semantic Versioning](https://sem
 ## Unreleased
 ### Changed
 ### Added
-- method to identify mask parameters with no TOAs and optionally freeze them
```
Comment (paulray) [inline CHANGELOG.md:10]: Why is this line deleted?

File: tests/test_fake_toas.py [hunk 2]
```diff
@@ -12,6 +12,34 @@
...
+    with tempfile.NamedTemporaryFile("wt") as f:
+        toas.write_TOA_file(f)
+        f.flush()
+        toas2 = get_TOAs(f.name, model=model)
+    return toas2
+
+
+def test_roundtrip():
+    # test for roundtrip accuracy at high precision
+    parfile = os.path.join(datadir, "NGC6440E.par")
```
Comment (paulray) [inline tests/test_fake_toas.py:25]: This roundtrip test is only done with NGC6440E.par, which has CLK UTC(NIST), which isn't even a valid CLK entry for  PINT. We should do a roundtrip test with a few other par files, with different settings for CLK including TT(TAI) and TT(BIPM2019) and maybe different PLANET_SHAPIRO as well

File: CHANGELOG.md (hunk 1 above)
Comment (dlakaplan) [reply CHANGELOG.md:10]: I don't know if that was intentional - changelog merges may have had issues

File: tests/test_fake_toas.py (hunk 2 above)
Comment (dlakaplan) [reply tests/test_fake_toas.py:25]: That's fair.  I think it failed before precisely because of the CLK value, but demonstrating that other clocks are OK is worthwhile.  I'll add that.

Comment (dlakaplan) [reply tests/test_fake_toas.py]: ok, I think the latest push does this

Comment (dlakaplan) [reply tests/test_fake_toas.py]: This does it for the clock/planet_shapiro combinations:
```
 ("UTC(NIST)", "Y"),
        ("UTC(NIST)", "N"),
        ("TT(BIPM2021)", "Y"),
        ("TT(BIPM2021)", "N"),
        ("TT(TAI)", "Y"),
        ("TT(TAI)", "N"),
```
happy to add others if wanted, but this already takes ~30s

Comment (paulray) [reply tests/test_fake_toas.py]: That seems sufficient! Could even remove one of the UTC(NIST) ones.

Comment (dlakaplan) [reply tests/test_fake_toas.py]: since the NIST one was the problematic one, I think it's best to keep it
